src/Resolvers/Collections.ts

```diff
@@ -92,31 +92,42 @@ export class CollectionsResolver {
 
   @FieldResolver(type => [Collection])
   async relatedCollections(
+    @Arg("size", () => Int, { nullable: true }) size: number,
     @Root() collection: Collection
   ): Promise<Collection[]> {
+    const query: { where: {}; take?: number } = { where: {} }
     const isArtistBased =
       collection.query.artist_ids && collection.query.artist_ids.length > 0
 
+    if (size) {
+      query.take = size + 1
+    }
+
     if (isArtistBased) {
-      const artistResults = await this.repository.find({
-        where: { "query.artist_ids": { $in: collection.query.artist_ids } },
-      })
+      query.where["query.artist_ids"] = {
+        $in: collection.query.artist_ids,
+      }
+
+      const artistResults = await this.repository.find(query)
 
       const relatedArtistCollections = reject(artistResults, {
         id: collection.id,
       })
 
+      // We only want to return relatedArtistCollections when there are more
```

Comment:
This is a really helpful comment, thank you.

Reply:
➕